asg/src/expression/circuit_init.rs

```diff
@@ -25,6 +25,7 @@ use std::cell::Cell;
 
 #[derive(Clone)]
 pub struct CircuitInitExpression<'a> {
+    pub id: u32,
```

Comment:
Definitely a fair suggestion!
As an alternative, could we wait until we discuss side-tables?

Reply:
I think `AsgId` already offers a benefit even before side-tables by making it harder to mix integers and improving readability. I would suggest eagerly new-typing integers in general. My experience in a compiler is that it often pays to take costs upfront rather than refactor later.

Reply:
Makes sense!